test/python/algorithms/quantum_time_evolution/variational/test_var_qite.py

```diff
@@ -0,0 +1,235 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test Variational Quantum Imaginary Time Evolution algorithm."""
+
+import unittest
+from test.python.algorithms import QiskitAlgorithmsTestCase
+import numpy as np
+
+from qiskit import Aer
+from qiskit.algorithms.quantum_time_evolution.variational.variational_principles.imaginary.implementations.imaginary_mc_lachlan_variational_principle import (
+    ImaginaryMcLachlanVariationalPrinciple,
+)
+from qiskit.algorithms.quantum_time_evolution.variational.var_qite import VarQite
+from qiskit.circuit.library import EfficientSU2
+from qiskit.opflow import (
+    SummedOp,
+    X,
+    Y,
+    I,
+    Z,
+)
+from qiskit.quantum_info import state_fidelity, Statevector
+
+np.random.seed = 11
+
+
+class TestVarQite(QiskitAlgorithmsTestCase):
+    """Test Variational Quantum Imaginary Time Evolution algorithm."""
+
+    def test_run_d_1(self):
+        """Test VarQite for d = 1 and t = 1."""
+        observable = SummedOp(
+            [
+                0.2252 * (I ^ I),
+                0.5716 * (Z ^ Z),
+                0.3435 * (I ^ Z),
+                -0.4347 * (Z ^ I),
+                0.091 * (Y ^ Y),
+                0.091 * (X ^ X),
+            ]
+        ).reduce()
+
+        d = 1
+        ansatz = EfficientSU2(observable.num_qubits, reps=d)
+
+        parameters = ansatz.ordered_parameters
+        init_param_values = np.zeros(len(ansatz.ordered_parameters))
+        for i in range(len(ansatz.ordered_parameters)):
+            init_param_values[i] = np.pi / 2
+        init_param_values[0] = 1
+        var_principle = ImaginaryMcLachlanVariationalPrinciple()
```

Comment:
Yes, I was also a bit unhappy about the redundancy. I will think about some pros and cons and we might discuss more, also with @Zoufalc.

Reply:
In the original code, I had a clear separation between VarQITE and VarQRTE as they were simply two different classes.
In this restructuring, I believe it would make sense to merge `VarQITE` and `VarQRTE` into `VarQTE` and then pass in the `ImaginaryMcLachlanVariationalPrinciple`.
I don't like the idea of setting real and imaginary keywords. This could be too confusing for the user. I'd rather keep a clear separation.

Reply:
Yeah we definitely do want a clear separation, I think the discussion is more about where we want it:

Option 1:
```
varqite_mclachlan = VarQITE(ImaginaryMcLachlanVariationalPrinciple(), ...)
```

Option 2:  
```
varqite_mclachlan = VarQITE(McLachlanVariationalPrinciple(), ...)
```

Option 3:
```
varqite_mclachlan = VarQTE(ImaginaryMcLachlanVariationalPrinciple(), ...)
```

Personally, I'd be for option 2 since that fits best our algorithm structure: all `QITE` algorithms are interchangeable, which wouldn't be the case for option 3. Option 1 would also fulfill that, but is more typing 😉

Reply:
I would agree with Christa's concerns about using keywords but also would not like to merge algorithms into `VarQte` per Julien's algorithm. Now I recall that was my reasoning to implement option 1 in the first place.

Reply:
But the user wouldn't have to set any keywords, that would be the job of the algorithm 🙂 Are there any variational principles that can only be applied to real or imaginary time evo?

Reply:
We decided to keep things as they are.